cmake/CMakeLists.txt

```diff
@@ -400,6 +402,57 @@ if(PKG_LATTE)
   list(APPEND LAMMPS_LINK_LIBS ${LATTE_LIBRARIES} ${LAPACK_LIBRARIES})
 endif()
 
+if(PKG_USER-SCAFACOS)
+  find_package(GSL REQUIRED)
+  option(DOWNLOAD_SCAFACOS "Download ScaFaCoS (instead of using the system's one)" OFF)
+  if(DOWNLOAD_SCAFACOS)
+    include(ExternalProject)
+    ExternalProject_Add(scafacos_build
+      URL https://github.com/scafacos/scafacos/releases/download/v1.0.1/scafacos-1.0.1.tar.gz
+      URL_MD5 bd46d74e3296bd8a444d731bb10c1738
+      CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> 
+                                               --disable-doc 
+                                               --enable-fcs-solvers=fmm,p2nfft,direct,ewald,p3m
+                                               --with-internal-fftw
+                                               --with-internal-pfft
+                                               --with-internal-pnfft
+                                               FC=${CMAKE_MPI_Fortran_COMPILER} 
+                                               CXX=${CMAKE_MPI_CXX_COMPILER} 
+                                               CC=${CMAKE_MPI_C_COMPILER}
+                                               F77=
+      )
+    ExternalProject_get_property(scafacos_build INSTALL_DIR)
+    set(SCAFACOS_BUILD_DIR ${INSTALL_DIR})
+    set(SCAFACOS_INCLUDE_DIRS ${SCAFACOS_BUILD_DIR}/include)
+    list(APPEND LAMMPS_DEPS scafacos_build)
+    # list and order from pkg_config file of ScaFaCoS build
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_direct.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_ewald.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_fmm.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_p2nfft.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_p3m.a)
+    list(APPEND LAMMPS_LINK_LIBS ${GSL_LIBRARIES})
+    list(APPEND LAMMPS_LINK_LIBS m)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_near.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_gridsort.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_resort.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_redist.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_common.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_pnfft.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_pfft.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_fftw3_mpi.a)
+    list(APPEND LAMMPS_LINK_LIBS ${SCAFACOS_BUILD_DIR}/lib/libfcs_fftw3.a)
```

Comment:
@rbberger do have a better idea than this explicit listing?

Reply:
@junghans not really, although hardcoding some of those (libm) seems wrong. We have that at some other place. Same goes for FFTW3, what is the benefit of having the internal one? Also, this breaks with `-D BUILD_LIB=on -D BUILD_SHARED_LIBS=on` due to missing `-fPIC`
```
/usr/bin/ld: scafacos_build-prefix/lib/libfcs_fmm.a(libfcs_fmm_back_idx_la-dash.o): relocation R_X86_64_PC32 against symbol `fcs_back_idx_sl_default_context' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/lammps.dir/build.make:6050: liblammps.so.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/lammps.dir/all] Error 2
```

Reply:
@rbberger, I fixed the `PIC` issue.

Reply:
@rbberger, there is a `--with-fftw3-libdir=DIR` and `--with-fftw3-includedir=DIR` option, so we could detect FFTW3 in CMake and hand this over to Scafacos if found or inject `--with-internal-fftw` if FFT3 isn't found.